src/tamp_improv/approaches/improvisational/policies/multi_rl.py

```diff
@@ -150,6 +190,10 @@ def _group_training_data(self, train_data: TrainingData) -> dict[str, TrainingDa
                     "states": [],
                     "current_atoms": [],
                     "preimages": [],
+                    "pattern": {
+                        "added_atoms": preimage - current_atoms,
```

Comment:
can we actually purge "preimage" to avoid confusion? we're not using preimages anymore, right?